src/main/java/microsoft/exchange/webservices/data/core/ExchangeServiceBase.java

```diff
@@ -193,8 +200,32 @@ private void initializeHttpClient() {
       .build();
   }
 
+  private void initializeHttpPoolingClient() {
+	    Registry<ConnectionSocketFactory> registry = createConnectionSocketFactoryRegistry();
+	    PoolingHttpClientConnectionManager httpConnectionManager = new PoolingHttpClientConnectionManager(registry);
+	    httpConnectionManager.setMaxTotal(maximumPoolingConnections);
+	    httpConnectionManager.setDefaultMaxPerRoute(maximumPoolingConnections);
+	    AuthenticationStrategy authStrategy = new CookieProcessingTargetAuthenticationStrategy();
+
+	    httpPoolingClient = HttpClients.custom()
+	      .setConnectionManager(httpConnectionManager)
+	      .setTargetAuthenticationStrategy(authStrategy)
+	      .build();
+	  }
+  
+  
   /**
-   * Create registry with configured {@link ConnectionSocketFactory} instances.
+   * Sets the maximum number of connections for the pooling connection manager which is used for subscriptions.
+   * <p>
+   * Default is 10.
+   * @param maximumPoolConnections
+   */
+  public void setMaximumPoolingConnections(int maximumPoolingConnections) {
+    this.maximumPoolingConnections = maximumPoolingConnections;
```

Comment:
maybe it might be usefull to include a check if the `maximumPoolingConnections` is >= 1 and throw an illegalArgument Exception if its not